Requirement Classification Clarification and other requirements Impro… (#50)
ietf-wg-nmop/draft-ietf-nmop-digital-map-concept: draft-ietf-nmop-simap-concept.md

# SIMAP Requirements

> The SIMAP requirements were split into 3 groups, for different target audience:

Reviewer: ```suggestion
The SIMAP requirements are split into three groups for different target audience:
```

> * Operator requirement:

Reviewer: ```suggestion
* Operator requirements:
```

> : These are requirements collected from the operators, they are functional requirements derived from the operators'

Reviewer: ```suggestion
: These requirements are collected from the operators. They are functional requirements derived from the operators'
```

> use cases. Some of the more specific semantic requirements were identified as RFC8345 gaps during the Hackaton with

Reviewer: ```suggestion
use cases. Some of the more specific semantic requirements were identified as {{!RFC8345}} gaps during the Hackathons with
```

> Operators and added as specific semantic requirements to the operator use cases.

Reviewer: ```suggestion
operators and added as specific semantic requirements to the operator use cases.
```

* Design requirements:

> : These are the requirements derived from the operator requirements and although there is some duplication,

Reviewer: ```suggestion
: These requirements are derived from the operators' requirements. Although there is some duplication,
```

> these are focused on summarizing the operator requirements for the design of the YANG module and API.

Reviewer: ```suggestion
these are focused on summarizing the operators' requirements for the design of the data model and API.
```

> These are functional requirements translated into the requirements for the model designer.

Reviewer: ```suggestion
These are functional requirements translated into low-level requirements for the model designers.
```

> The reason for doing this is to ensure that the YANG module is designed according to the operator

Reviewer: ```suggestion
The rationale for adopting this approach is to ensure that the data model is designed according to the operators'
```

> requirements, and that they could be used for both design and review of the proposed YANG module(s).

Reviewer: ```suggestion
requirements and that they could be used for both design and review of the candidate YANG module(s).
```

> * Architecture Requirements:

Reviewer: ```suggestion
* Architecture requirements:
```

> : We added architectural non-functional requirements as well, as operators were mentioning performance, scale and

Reviewer: ```suggestion
: Architectural (non-functional) requirements are captured as well, as operators identified performance needs, large scale support,  and
```

> network discovery. These are not requirements for the YANG Module, but are requirements either for API itself or for

Reviewer: ```suggestion
network discovery. These are not requirements for the data model, but are requirements either to drive the API design itself (e.g., to better optimize performance) or for
```

> the controllers/orchestrators that provide the SIMAP API. Although, they may be common sense requirements not specific

Reviewer: ```suggestion
the network controllers and orchestrators that expose a SIMAP API. Although, they may be common sense requirements not specific
```

> to SAIN API, we decided to list them here as well.

Reviewer: ```suggestion
to SIMAP API,  these are list here for completeness.
```

## Operator Requirements

> The following are the operator requirements for the SIMAP (note that some of them are supported by

Reviewer: ```suggestion
The following are the operators' requirements for the SIMAP. Note that some of these requirements are supported by
```

> default by {{!RFC8345}}):

Reviewer: ```suggestion
default by {{!RFC8345}}.
```

The following requirements are more specific requirements for semantics

REQ-LAYER-NAVIGATE:
: SIMAP must provide capability to navigate inside the topology layer and between the topology layers

REQ-EXTENSIBLE:

> : SIMAP must be extensible with metadata.

Reviewer: We need to define what is metadata in our context.

REQ-PLUGG:
: SIMAP must be pluggable. That is,

     + Must connect to other YANG modules for device configuration, inventory, configuration, assurance, etc.

> The SIMAP does not contain the detailed configuration for the device, so we should be able to link to it from SIMAP.

Reviewer: ```suggestion
The SIMAP does not contain the detailed device configuration, so a mechanism is needed to be able to link it from SIMAP.
```

SIMAP should also be linked to a 'logical configuration inventory'. Several examples of the type of logical information

> to be linked from SIMAP: inventory of logical interfaces, inventory of ACLs, inventory of routing policies.

Reviewer: ```suggestion
to be linked from SIMAP: inventory of logical interfaces, inventory of ACLs, or inventory of routing policies.
```

: SIMAP entities should mainly contain properties used to identify topological entities at different layers,
identify their roles, and topological relationships between them.

: SIMAP entities should also provide information required to define semantics for layered network topologies, such as:

> * link directionality

Reviewer: ```suggestion
* link directionality,
```
